app/controllers/api/v2/job_invocations_controller.rb

```diff
@@ -214,7 +214,7 @@ def output_lines_since(task, time)
       end
 
       def host_output(job_invocation, host, default: nil, since: nil, raw: false)
-        refresh = true
+        refresh = !job_invocation.finished?
```

Comment:
It would be good to be more defensive here, in case the task does not exist yet, `finished?` could fail since it calls `task.pending`. I admit the fix may need to land in `def finished?`.